go/pfconfigdriver/pool_test.go

```diff
@@ -0,0 +1,136 @@
+package pfconfigdriver
+
+import (
+	"github.com/fingerbank/processor/sharedutils"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+)
+
+func TestLoadResource(t *testing.T) {
+	rp := NewResourcePool(ctx)
+	gen := PfConfGeneral{}
+
+	// Test loading a resource and validating the result
+	loaded, err := rp.LoadResource(ctx, &gen, true)
+	sharedutils.CheckTestError(t, err)
+
+	if !loaded {
+		t.Error("Resource wasn't loaded when calling a first time load")
+	}
+
+	expected := "pfdemo.org"
+	if gen.Domain != expected {
+		t.Errorf("Resource domain wasn't loaded correctly through resource pool. Got %s instead of %s", gen.Domain, expected)
+	}
+
+	query := createQuery(ctx, &gen)
+	_, ok := rp.loadedResources[query.GetIdentifier()]
+	if !ok {
+		t.Error("The loaded resource wasn't stored in the pool")
+		return
+	}
+
+	// Test loading a resource with the firstLoad flag which should reload from pfconfig even though there is another resource that uses the same struct
+	gen = PfConfGeneral{}
+	loaded, err = rp.LoadResource(ctx, &gen, true)
+	sharedutils.CheckTestError(t, err)
+
+	if !loaded {
+		t.Error("Resource wasn't loaded when calling a first time load")
+	}
+
+	// Test loading a resource without the firstLoad flag which shouldn't read from pfconfig
+	gen = PfConfGeneral{}
+	loaded, err = rp.LoadResource(ctx, &gen, true)
+	sharedutils.CheckTestError(t, err)
+
+	if !loaded {
+		t.Error("Resource wasn't loaded when calling a first time load")
+	}
+
+	expected = "pfdemo.org"
+	if gen.Domain != expected {
+		t.Errorf("Resource domain wasn't loaded correctly through resource pool. Got %s instead of %s", gen.Domain, expected)
+	}
+
+	// Test changing data in pfconfig and reloading the resource
+	cmd := exec.Command("sed", "-i.bak", "s/domain=pfdemo.org/domain=zammitcorp.com/g", "/usr/local/pf/t/data/pf.conf")
```

Comment:
Needed a way to modify the config file without the packetfence perl libraries.

That was the best quick way I found

Don't worry, I restore the actual test file after the test